src/Modules/local_rings.jl

```diff
@@ -0,0 +1,41 @@
+export length
+export decomposition_series
+
+function length(M::ModuleFP{RingElemType}) where {RingElemType<:AbsLocalizedRingElem{<:Any, <:Any, <:MPolyComplementOfPrimeIdeal}}
+  if iszero(M) 
+    set_attribute!(M, :decomposition_series, Vector{elem_type(M)}())
+    return 0
+  end
+  R = base_ring(M)
+  P = prime_ideal(inverted_set(R))
+  F = FreeMod(R, 1)
+  N, _ = quo(F, (R(P)*F)[1])
+  k = 0
+  MM = M
+  p = identity_map(M)
+  NtoM, interp = hom(N, MM)
```

Comment:
interp is a non-intuitive name for this map, but I do not have a better name off-hand either.